R/agyw-integration.R

```diff
@@ -0,0 +1,1638 @@
+
+  rr_reg_dat <- data.frame(genpop_incidence = df$incidence/100) %>%
+    dplyr::mutate(log_gen = log(genpop_incidence),
+                  log_sexpaid12m = 0.604104017 * log_gen + 0.075090952,
+                  sexpaid12m_incidence = exp(log_sexpaid12m),
+                  rr_sexpaid12m = sexpaid12m_incidence / genpop_incidence)
+
+  rr_sexpaid12m <- rr_reg_dat$rr_sexpaid12m
+  # This gives implausibly high RRs for very low districts (e.g. IRR = 297!)
+  # capping at 35
+  rr_sexpaid12m[rr_sexpaid12m > 35] <- 35
+
+  # TODO: Get distributions on these and using a sampling method to get
+  # uncertainty in economic analysis e.g.
+  rr_sexnonreg_se <- 0.2
+  rr_sexnonreg_se <- 1
+
+
+  # Calculate risk group incidence
+  Y015_024 <-  c("Y015_019", "Y020_024")
+  Y025_049 <-  c("Y025_029","Y030_034","Y035_039","Y040_044", "Y045_049")
+
+  df1 <- df %>%
+    dplyr::mutate(
+      rr_sexpaid12m = rr_sexpaid12m,
+      rr_sexnonreg = dplyr::case_when(
+        age_group %in% Y015_024 ~ rr_sexnonreg_young,
+        age_group %in% Y025_049 ~ rr_sexnonreg_old,
+        TRUE ~ NA_real_),
+      population_nosex12m = population * nosex12m,
+      population_sexcohab = population * sexcohab,
+      population_sexnonreg = population * sexnonreg,
+      population_sexpaid12m = population * sexpaid12m,
+      plhiv_nosex12m = population_nosex12m * prev_nosex12m,
+      plhiv_sexcohab = population_sexcohab * prev_sexcohab,
+      plhiv_sexnonreg = population_sexnonreg * prev_sexnonreg,
+      plhiv_sexpaid12m = population_sexpaid12m * prev_sexpaid12m,
+      susceptible_nosex12m = population_nosex12m - plhiv_nosex12m,
+      susceptible_sexcohab = population_sexcohab - plhiv_sexcohab,
+      susceptible_sexnonreg = population_sexnonreg - plhiv_sexnonreg,
+      susceptible_sexpaid12m = population_sexpaid12m - plhiv_sexpaid12m,
+      incidence_sexpaid12m = (incidence/100) * rr_sexpaid12m,
+      infections_sexpaid12m = susceptible_sexpaid12m * incidence_sexpaid12m,
+      incidence_nosex12m = 0,
+      incidence_sexcohab = (infections - infections_sexpaid12m) / (susceptible_sexcohab + rr_sexnonreg * susceptible_sexnonreg),
+      incidence_sexnonreg = incidence_sexcohab * rr_sexnonreg,
+      infections_nosex12m = 0,
+      infections_sexcohab = susceptible_sexcohab * incidence_sexcohab,
+      infections_sexnonreg = susceptible_sexnonreg * incidence_sexnonreg)
+
+  # Calculate risk group incidence for aggregate age groups
+
+  summarise_age_cat_female <- function(dat, age_cat) {
+
+    if (age_cat == "Y015_024") {age_groups <- c("Y015_019", "Y020_024")}
+    if (age_cat == "Y025_049") {age_groups <- c("Y025_029","Y030_034","Y035_039",
+                                                "Y040_044", "Y045_049")}
+    if (age_cat == "Y015_049") {age_groups <- c("Y015_019", "Y020_024","Y025_029",
+                                                "Y030_034","Y035_039","Y040_044",
+                                                "Y045_049")}
```

Comment:
Similar comments to above; would prefer to avoid hardcoding these; fine to leave for now.